Assets/Scripts/Controllers/HeaderController.cs

```diff
@@ -12,22 +13,45 @@ public class HeaderController : MonoBehaviour {
 		public HeaderConfig Config;
 
 		private Vector3 targetPosition;
+		private Vector3 targetPrimaryRangePosition, targetSecondaryRangePosition;
+		private Vector2 targetPrimaryRangeSize, targetSecondaryRangeSize;
+		private SpriteRenderer headerIndicatorPrimaryRangeSprite, headerIndicatorSecondaryRangeSprite;
+		private float indicatorWidth, indicatorHeight;
+		private int currentStart, currentEnd, currentCount;
+
 		private InputController inputController;
 		private NodeController nodeController;
 		private NetworkController networkController;
+		private ConnectionController connectionController;
 
 		void Start() {
 			if (inputController.Environment == Environment.PC) {
 				HeaderObject.transform.parent = Camera.main.transform;
 			}
 			nodeController.OnSelectedNodeChanged += UpdateNodeHeaderAfterSelectOrHighlight;
 			nodeController.OnHighlightedNodeChanged += UpdateNodeHeaderAfterSelectOrHighlight;
+			connectionController.OnConnectionRangeChanged += UpdateNodeHeaderAfterConnectionRangeChange;
+			connectionController.OnConnectionRangeChanged?.Invoke(0, 0, 0);
+
+			Transform indicatorBase = HeaderObject.transform.GetChild(3);
+			indicatorBase.GetComponent<SpriteRenderer>().size = Config.HeaderRangeSize;
+			indicatorWidth = indicatorBase.GetComponent<SpriteRenderer>().size.x;
+			indicatorHeight = indicatorBase.GetComponent<SpriteRenderer>().size.y;
+			headerIndicatorPrimaryRangeSprite = indicatorBase.GetChild(0).GetComponent<SpriteRenderer>();
+			headerIndicatorSecondaryRangeSprite = indicatorBase.GetChild(1).GetComponent<SpriteRenderer>();
+
+			// Render always on top of nodes and connections
+			HeaderObject.transform.GetChild(0).GetComponent<MeshRenderer>().sortingOrder = 50;
+			HeaderObject.transform.GetChild(1).GetComponent<MeshRenderer>().sortingOrder = 50;
+			HeaderObject.transform.GetChild(2).GetComponent<MeshRenderer>().sortingOrder = 50;
```

Comment:
- Try to extract repeated code to functions
- declare private readonly constant fields if necessary

Reply:
Don't know if that's better. 
This is only there, because I can't set this parameter through Unity`s Inspector.
![obraz](https://user-images.githubusercontent.com/36575365/68073331-369e3780-fd8f-11e9-8d7b-2ea8453c70c9.png)
![obraz](https://user-images.githubusercontent.com/36575365/68073330-3140ed00-fd8f-11e9-8a80-6f44885e7f17.png)